src/test/java/me/gnat008/perworldinventory/config/SettingsConsistencyTest.java

```diff
@@ -17,37 +25,110 @@
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
 
 /**
  * Tests that the config.yml file corresponds with the settings holder classes in the code.
  */
 public class SettingsConsistencyTest {
 
     /** Bukkit's FileConfiguration#getKeys returns all inner nodes also. We want to exclude those in tests. */
-    private static final List<String> YAML_INNER_NODES = ImmutableList.of("player", "player.stats");
+    private static final List<String> YAML_INNER_NODES = ImmutableList.of("player", "player.stats", "database");
+    private static final String SETTINGS_FOLDER = TestHelper.PROJECT_ROOT + "config";
 
-    private final ConfigurationData configData = ConfigurationDataBuilder.collectData(PwiProperties.class);
-    private final FileConfiguration ymlConfiguration = YamlConfiguration.loadConfiguration(getJarFile("/config.yml"));
+    private static ConfigurationData configData;
+    private static YamlFileResource yamlResource;
+    private static List<Class<? extends SettingsHolder>> classes;
 
+    @BeforeClass
+    public static void scanForSettingsClasses() {
+        ClassCollector collector = new ClassCollector(TestHelper.SOURCES_FOLDER, SETTINGS_FOLDER);
+        classes = collector.collectClasses(SettingsHolder.class);
+
+        if (classes.isEmpty()) {
+            throw new IllegalStateException("Did not find any SettingsHolder classes. Is the folder correct?");
+        }
+
+        System.out.println("Found " + classes.size() + " SettingsHolder implementations");
+
+        configData = SettingsRetriever.buildConfigurationData();
```

Comment:
You just scanned for all SettingsHolder classes -- use them to build your configData. `ConfigurationDataBuilder.collectData(classes)`